src/ert/storage/local_ensemble.py

```diff
@@ -275,67 +276,17 @@ def load_gen_kw_realization(
 
         return np_data, keys
 
-    def load_summary_data_as_df(
-        self, summary_keys: List[str], realizations: List[int]
-    ) -> pd.DataFrame:
-        result = []
-        key_set: Set[str] = set()
-        for realization in realizations:
-            input_path = (
-                self.mount_point / f"realization-{realization}" / "summary-data.nc"
-            )
-            if not input_path.exists():
-                continue
-            with xr.open_dataset(input_path, engine="scipy") as ds_disk:
-                result.append(ds_disk)
-                if not key_set:
-                    key_set = set(sorted(ds_disk["data_key"].values))
-        if not result:
-            raise KeyError(f"Unable to load SUMMARY_DATA for keys: {summary_keys}")
-        df = xr.merge(result).to_dataframe(dim_order=["data_key", "axis"])
-        # realization nr is stored as str in netcdf
-        dropped_keys = key_set - set(summary_keys)
-        if len(dropped_keys) > 0:
-            df.drop(dropped_keys, level="data_key", inplace=True)
-        df.columns = [int(x) for x in df.columns]
-        return df
-
-    def load_gen_data(
-        self, key: str, realizations: List[int]
-    ) -> Tuple[npt.NDArray[np.double], List[int]]:
-        result = []
-        loaded = []
-        for realization in realizations:
-            input_path = self.mount_point / f"realization-{realization}"
-            if not input_path.exists():
-                continue
-
-            with xr.open_dataset(input_path / "gen-data.nc", engine="scipy") as ds_disk:
-                np_data = ds_disk[key].as_numpy()
-                result.append(np_data)
-                loaded.append(realization)
-        if not result:
-            raise KeyError(f"Unable to load GEN_DATA for key: {key}")
-        return np.stack(result).T, loaded
-
-    def load_gen_data_as_df(
-        self, keys: List[str], realizations: List[int]
-    ) -> pd.DataFrame:
-        dfs = []
-        for key in keys:
-            data, realizations = self.load_gen_data(key, realizations)
-            x_axis = [*range(data.shape[0])]
-            multi_index = pd.MultiIndex.from_product(
-                [[key], x_axis], names=["data_key", "axis"]
-            )
-            dfs.append(
-                pd.DataFrame(
-                    data=data,
-                    index=multi_index,
-                    columns=realizations,
-                )
-            )
-        return pd.concat(dfs)
+    @lru_cache
+    def load_response(self, key: str, realizations: Tuple[int, ...]) -> xr.Dataset:
+        realizations = [
+            (self.mount_point / f"realization-{i}/{key}/responses.nc")
+            for i in realizations
+            if (self.mount_point / f"realization-{i}/{key}/responses.nc").exists()
+        ]
+        if not realizations:
+            return xr.Dataset()
+        response = xr.open_mfdataset(realizations, engine="scipy")
```

Comment:
Did you try with `parallel=True`?

From the docs:

parallel ([bool](https://docs.python.org/3/library/functions.html#bool), default: [False](https://docs.python.org/3/library/constants.html#False)) – If True, the open and preprocess steps of this function will be performed in parallel using dask.delayed. Default is False.

Reply:
Yes, tried it, did not see a performance boost, granted my testing was a single (large) response, and not on NFS.

Reply:
Checked it again, with 500 realizations and a single response with 100 values. Was quite a bit slower with parallel.